src/main/java/datawave/query/util/FrequencyFamilyCounter.java

```diff
@@ -0,0 +1,135 @@
+    private static final int SIMPLEDATE_LENGTH = 8;
+    private DateFrequencyValue serializer = new DateFrequencyValue();
+    
+    private static final Logger log = LoggerFactory.getLogger(FrequencyFamilyCounter.class);
+    
+    public FrequencyFamilyCounter() {
+        
+    }
+    
+    public void initialize(Value value) {
+        deserializeCompressedValue(value);
+    }
+    
+    public void clear() {
+        dateToFrequencyValueMap.clear();
+    }
+    
+    public TreeMap<YearMonthDay,Frequency> getDateToFrequencyValueMap() {
+        return dateToFrequencyValueMap;
+    }
+    
+    /**
+     * Takes the value of a compressed f column record and creates the qualifierToFrequencyValueMap so that other records that are not yet aggregated can be
+     * added to the compressed record. After those records are aggregated they are discarded.
+     *
+     * @param oldValue
+     */
+    public void deserializeCompressedValue(Value oldValue) {
+        dateToFrequencyValueMap = serializer.deserialize(oldValue);
+    }
+    
+    /**
+     * Performs what the SummingCombiner used to perform for date and frequency values. The function accumulutes ingest frequencies for individual dates
+     */
+    
+    public void aggregateRecord(String key, String value) {
+        
+        insertIntoMap(key, value);
+    }
+    
+    /**
+     * Inserts a key and value into the qualifiedToFrequencyValueMap and converts the string value to a long
+     * 
+     * @param key
+     *            The value for "key" is the old style ColumnQualifier of this format: csv\x0020160426
+     * @param value
+     *            value is a an Integer in string format that might be hexadecimal, decimal or octal
+     */
+    public void insertIntoMap(String key, String value) {
+        int parsedLong;
+        String cleanKey;
+        
+        // Assuming that as SimpleDate is at the end of the key passed in. yyyyMMdd
+        // The value for "key" is the old style ColumnQualifier of this format:
+        // csv\x0020160426 and the value is a an Integer in string format that
+        // might be hexadecimal, decimal or octal
+        if (key != null) {
+            if (key.length() > SIMPLEDATE_LENGTH) {
+                cleanKey = key.substring(key.length() - SIMPLEDATE_LENGTH);
+            } else {
+                cleanKey = key;
+            }
+        } else
+            return;
+        
+        if (log.isTraceEnabled())
+            log.trace("inserting key: " + cleanKey + " value: " + value);
+        
+        if (value.isEmpty())
+            return;
+        
+        try {
+            parsedLong = Integer.parseUnsignedInt(value);
+        } catch (NumberFormatException nfe) {
+            try {
+                if (log.isTraceEnabled()) {
+                    log.trace("Long.parseLong could not parse " + value + " to long for this key " + cleanKey);
+                    log.trace("Trying to use Long.decode");
```

Comment:
I see Integer.decode, not Long.decode here....

Reply:
Back when we were getting started on this you told me the highest frequency value we could reasonable expect was java.language.Integer.MAX_VALUE = 2147483647 .   That would take 4 bytes to hold which is why the encoding class only encodes/decodes 4 bytes.  Long.decode may requires up to 8 bytes which is double the memory footprint.   We can go to 8 bytes of allocation for frequency if you want.  Let me know.

Reply:
My point was that your log.trace statement is either incorrect or your statement on line 99 is incorrect.  Please reconcile these lines.